mono/mini/mini-windows.c

```diff
@@ -444,3 +445,29 @@ mono_thread_state_init_from_handle (MonoThreadUnwindState *tctx, MonoThreadInfo
 
 	return TRUE;
 }
+
+BOOL
+mono_win32_runtime_tls_callback (HMODULE module_handle, DWORD reason, LPVOID reserved, MonoWin32TLSCallbackType callback_type)
+{
+	if (!mono_win32_handle_tls_callback_type (callback_type))
+		return TRUE;
+
+	if (!mono_gc_dllmain (module_handle, reason, reserved))
+		return FALSE;
+
+	switch (reason)
+	{
+	case DLL_PROCESS_ATTACH:
+		mono_install_runtime_load (mini_init);
+		break;
+	case DLL_PROCESS_DETACH:
+		if (coree_module_handle)
+			FreeLibrary (coree_module_handle);
```

Comment:
Probably should check `reserved` and do nothing if it indicates ExitProcess, instead of FreeLibrary.
It isn't actually `reserved. It has a documented meaning.
Cleanup within ExitProcess tends not to work.